API/Controllers/CategoryController.cs

```diff
@@ -2,10 +2,72 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using API.Interfaces;
+using API.Data;
+using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using Microsoft.AspNetCore.Mvc;
+using API.DTOs;
+using API.Extensions;
+using API.Entities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using System.Security.Claims;
 
 namespace API.Controllers
 {
+    [Authorize]
     public class CategoryController : BaseApiController
     {
+        private readonly ICategoryRepository _categoryRepository;
+        private readonly IMapper _mapper;
+        private readonly IUserRepository _userRepository;
+
+        public CategoryController(ICategoryRepository categoryRepository,
+            IMapper mapper, IUserRepository userRepository)
+        {
+            _categoryRepository = categoryRepository;
+            _mapper = mapper;
+            _userRepository = userRepository;
+        }
+
+        [HttpPost]
+        public async Task<ActionResult<Category>> CreateCategory(Category category)
```

Comment:
Nie używaj encji jako argumentów lub zwracanego typu w kontrolerze . Po to masz dto. Teraz jak coś zmienisz w encji, dodasz nowe pole to ci się zacznie doklejać do zwracanych jsonów itd.